nilearn/plotting/html_surface.py

```diff
@@ -21,25 +21,45 @@ class SurfaceView(HTMLDocument):
 
 
 def _get_vertexcolor(surf_map, cmap, norm,
-                     absolute_threshold=None, bg_map=None):
+                     absolute_threshold=None, bg_map=None,
+                     bg_on_data=None, scale_bg_map=None, darkness=None):
     vertexcolor = cmap(norm(surf_map).data)
-    if absolute_threshold is None:
-        return to_color_strings(vertexcolor)
+
     if bg_map is None:
         bg_map = np.ones(len(surf_map)) * .5
         bg_vmin, bg_vmax = 0, 1
     else:
         bg_map = surface.load_surf_data(bg_map)
+
+    # scale background map if need be
+    if scale_bg_map:
         bg_vmin, bg_vmax = np.min(bg_map), np.max(bg_map)
-    bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
-    bg_color = plt.get_cmap('Greys')(bg_norm(bg_map))
-    vertexcolor[np.abs(surf_map) < absolute_threshold] = bg_color[
-        np.abs(surf_map) < absolute_threshold]
+        bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
+        bg_data = bg_norm(bg_map)
+    else:
+        bg_data = bg_map
+
+    if darkness is not None:
+        bg_data *= darkness
+    bg_color = plt.get_cmap('Greys')(bg_data)
+
+    # select vertices which are filtered out by the threshold
```

Comment:
Any chance we could add some tests for `_get_vertexcolor` with different values for `scale_bg_map` ? 

There are already a few things here:

https://github.com/nilearn/nilearn/blob/0bba4b600e164f2f3cd209853c632e1ca635ec8b/nilearn/plotting/tests/test_html_surface.py#L20

Reply:
I added two tests with `scale_bg_map` equal to `False` and `True` respectively!